### PR TITLE
- Added rotating tips. 5 tips that cycle every 60 seconds with fadeout/fadein

### PR DESCRIPTION
**Pull request by PureKrome**
5 Tips :-
- Type @ then press TAB to auto-complete nicknames
- Type /help to see the list of commands
- Type /rooms to list all available rooms
- Type : then press TAB to auto-complete emoji icons
- You can create your own private rooms. Type /help for more info

Fade Out / Fade In = 2 secs each.

Rotate is set for every 60 seconds.

*\* Apologies for the 2nd commit. I wanted to get my style, right.

### CODE REVIEW

File: JabbR/index.htm [hunk 1]
```diff
@@ -43,6 +43,33 @@
...
+                    currentTipIndex++;
+                    if (currentTipIndex >= tips.length) {
+                        currentTipIndex = 0;
+                    }
+                    $('#message-instruction').fadeOut(2000, function () {
+                        $('#message-instruction').html(tips[currentTipIndex]);
+                    });
+
+                    $('#message-instruction').fadeIn(2000, recursiveTimeout());
+                }, 60 * 1000);
```
Comment (davidfowl) [inline JabbR/index.htm]: Make this a constant.

File: JabbR/index.htm [hunk 2]
```diff
@@ -43,6 +43,33 @@
...
+                'Type @ then press TAB to auto-complete nicknames',
+                'Type /help to see the list of commands',
+                'Type /rooms to list all available rooms',
+                'Type : then press TAB to auto-complete emoji icons',
+                'You can create your own private rooms. Type /help for more info'
+            ];
+
+            var currentTipIndex = 0;
+
+            function recursiveTimeout() {
```
Comment (davidfowl) [inline JabbR/index.htm]: Make this name more meaningful. cycleTips.

File: JabbR/index.htm (hunk 1 above)
Comment (PureKrome) [inline JabbR/index.htm]: do you mean, the time in milliseconds?

Comment (davidfowl) [inline JabbR/index.htm]: something readable.

**Comment by PureKrome** [comment]
Updated.

File: JabbR/index.htm [hunk 3]
```diff
@@ -43,6 +43,34 @@
...
+                'Type /help to see the list of commands',
+                'Type /rooms to list all available rooms',
+                'Type : then press TAB to auto-complete emoji icons',
+                'You can create your own private rooms. Type /help for more info'
+            ];
+
+            var cycleTimeInMilliseconds = 60 * 1000; // 1 minute.
+            var currentTipIndex = 0;
+            
+            function recursiveTimeout() {
```
Comment (davidfowl) [inline JabbR/index.htm]: Change name to cycleMessages

Comment (PureKrome) [inline JabbR/index.htm]: Done.

File: JabbR/index.htm [hunk 4]
```diff
@@ -43,6 +43,34 @@
...
+                'Type /help to see the list of commands',
+                'Type /rooms to list all available rooms',
+                'Type : then press TAB to auto-complete emoji icons',
+                'You can create your own private rooms. Type /help for more info'
+            ];
+
+            var cycleTimeInMilliseconds = 60 * 1000; // 1 minute.
+            var cycleMessageIndex = 0;
+            
+            function recursiveTimeout() {
```
Comment (davidfowl) [inline JabbR/index.htm]: Change the name of recursiveTimeout to cycleMessages...

Comment (PureKrome) [inline JabbR/index.htm]: Updated. 
(sorry about the slow update -> weekends, kids, no life, QQ....)

**Comment by davidfowl** [comment]
Why is this code in index.htm?

**Comment by PureKrome** [comment]
3 reasons.
1. The message div is located in this file.
2. it's tiny.
3. i didn't know of any other js file where it would be appropriate to place it, in. Having it in a seperate js file is preferable because of caching.

I would prefer to have it in a js file. Any preference? chat.ui.js?

**Comment by davidfowl** [comment]
Yes move it to the ui file :).

**Comment by PureKrome** [comment]
Ok. done.

few points to note:
1. I'm kick starting the function logic by calling the jquery `ready` logic. I'm not 100% sure if this is the best place to do it, so I'm happy to move this starting call to a more appropriate place (i'm just not 100% familiar with the js jabbr code).

```
// Start cycling the messages once the document has finished loading.
$(document).ready(cycleMessages());
```
1. Even though i've added a variable to define the cycle time in milliseconds, I'm not sure if that's an overkill. So i'm also happy to replace this..

`var cycleTimeInMilliseconds = 60 * 1000; // 1 minute.
 $('#message-instruction').fadeIn(2000, cycleMessages());
}, cycleTimeInMilliseconds);`

with this

`$('#message-instruction').fadeIn(2000, cycleMessages());
}, 60000);`

File: JabbR/Chat.ui.js [hunk 5]
```diff
@@ -654,6 +654,35 @@
...
+            $('#message-instruction').fadeOut(2000, function () {
+                $('#message-instruction').html(messages[messageIndex]);
+            });
+
+            $('#message-instruction').fadeIn(2000, cycleMessages());
+        }, cycleTimeInMilliseconds);
+    };
+
+    // Start cycling the messages once the document has finished loading.
+    $(document).ready(cycleMessages());
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: Yea this is all wrong and doesnt follow the pattern we use today. Kick it off in ui.initialize.

Comment (PureKrome) [inline JabbR/Chat.ui.js]: Updated. (added to the bottom of the ui.init.  I didn't know that function existed. Awesome!)

**Pull request merged**